File: server-java/src/main/java/com/zhiyu/health/entity/knowledge/KnowledgeDocument.java
package com.zhiyu.health.entity.knowledge;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.OffsetDateTime;
import lombok.Getter;
import lombok.Setter;

/**
 * 知识文档元数据与异步处理状态机（票 89，ADR-0036）。
 *
 * <p>source=SEED 为系统预置（只读，永远 READY），source=UPLOAD 为运营上传。
 * status 四态 PROCESSING/READY/FAILED/ARCHIVED；原文旁路存 MinIO（object_key 降级空）。
 * processing_started_at 用于启动时孤儿恢复扫描（超时标 ORPHANED）。
 */
@Getter
@Setter
@TableName("knowledge_documents")
public class KnowledgeDocument {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String fileName;
    private String contentType;
    private Long byteSize;
    private String objectKey;
    private String source;
    private String status;
    private String department;
    private Long uploaderStaffId;
    private String errorCode;
    private String errorMessage;
    private OffsetDateTime processingStartedAt;
    private Integer chunkCount;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: admin/src/services/knowledgeDocument.ts
import { request } from '@umijs/max';
import { getToken } from '@/utils/session';

export interface KnowledgeDocument {
  id: number;
  file_name: string;
  content_type: string;
  byte_size: number;
  source: 'SEED' | 'UPLOAD';
  status: 'PROCESSING' | 'READY' | 'FAILED' | 'ARCHIVED';
  department: string | null;
  chunk_count: number;
  error_code: string | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface UploadResponse {
  id: number;
  status: string;
}

export function listKnowledgeDocuments() {
  return request<KnowledgeDocument[]>('/api/b/knowledge-documents');
}

export function retryKnowledgeDocument(id: number) {
  return request<UploadResponse>(`/api/b/knowledge-documents/${id}/retry`, { method: 'POST' });
}

export function archiveKnowledgeDocument(id: number) {
  return request<UploadResponse>(`/api/b/knowledge-documents/${id}/archive`, { method: 'POST' });
}

// antd Upload 的 customRequest 不走 Umi request 拦截器，须手动带 Bearer
export function uploadKnowledgeDocument(file: File, department: string): Promise<UploadResponse> {
  const form = new FormData();
  form.append('file', file);
  form.append('department', department);
  return fetch('/api/b/knowledge-documents', {
    method: 'POST',
    headers: { Authorization: `Bearer ${getToken()}` },
    body: form,
  }).then(async (resp) => {
    if (!resp.ok) {
      const detail = await resp.json().catch(() => ({}));
      throw new Error(typeof detail.detail === 'string' ? detail.detail : '文档上传失败');
    }
    return resp.json() as Promise<UploadResponse>;
  });
}
